util/reggen/validate.py

```diff
@@ -1296,25 +1300,28 @@ def check_wen_regs(regs):
     # 0 - name
     # 1 - reset value
     # 2 - sw access
+    # 3 - sw access
     tuple_name = 0
     tuple_rstval = 1
     tuple_swaccess = 2
+    tuple_hwaccess = 3
 
-    reg_list = [(reg['name'].lower(), reg['genresval'], reg['swaccess'])
-                for reg in regs['registers']
+    reg_list = [(reg['name'].lower(), reg['genresval'], reg['swaccess'],
+                 reg['hwaccess']) for reg in regs['registers']
                 if 'name' in reg and 'genresval' in reg]
     mreg_list = [
         reg['multireg'] for reg in regs['registers'] if 'multireg' in reg
     ]
     field_list = [((reg['name'] + "_" + field['name']).lower(),
-                   field['genresval'], field['swaccess']) for mreg in mreg_list
-                  for reg in mreg['genregs'] for field in reg['fields']]
+                   field['genresval'], field['swaccess'], field['hwaccess'])
+                  for mreg in mreg_list for reg in mreg['genregs']
+                  for field in reg['fields']]
```

Comment:
why does this indentation or lack thereof look strange? it appears you've run this through yapf